zebra-consensus/src/transaction.rs

```diff
@@ -570,7 +676,41 @@ where
                 continue;
             }
         }
-        Ok((spent_utxos, spent_outputs))
+
+        if let Some(mempool) = mempool {
```

Comment:
This new implementation changes the lookup logic. 

In the old implementation, if `is_mempool` was set, and `UnspentBestChainUtxo` returned `None`, the fn returned `TransactionError::TransparentInputNotFound`.

In the new implementation, if the same conditions hold, and `mempool` is `None`, no error is returned.

Reply:
Oh no, the above is wrong, closing.